src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styleDeclarations = sourceString
+    .split(';')
+    .map((style) => style.trim())
+    .filter((style) => style.length > 0);
+
+  return styleDeclarations.reduce((acc, style) => {
+    const [key, value] = style.split(':');
```

Comment:
[CODE KNOWLEDGE] - There's no need to make a copy before each operation. Methods like `split` do not mutate the original string, so you can chain them directly without using `map` to trim each style.